ITE/4/README.adoc

```diff
@@ -0,0 +1,361 @@
+        "workdir": ""
+    }
+}
+```
+
+This step is accepting the pull request as a material and is recording the merge
+commit as a product.
+
+===== in-toto link attesations for GitHub Actions building from a merge commit
+
+GitHub Actions can be used to set up a workflow for continuous integration (CI).
+Workflows can be triggered on push and an attestation can be generated for the
+build that as a result.
+
+
+```
+{
+    "_type": "link",
+    "name": "github-actions-build-pull-request-250",
+    "command": "",
+    "materials": {
+        "github+https://github.com/in-toto/in-toto/commit/f1c5d201887e226cadac5792a203ac3eae347add": <HASH>
+    },
+    "products": {
+        "github+https://github.com/in-toto/in-toto/commit/f1c5d201887e226cadac5792a203ac3eae347add/checks?check_suite_id=<ID>": <HASH>
+    },
+    "byproducts": {
+        "stderr": "",
+        "stdout": "",
+        "return-value": ""
+    },
+    "environment": {
+        "variables": "",
+        "filesystem": "",
+        "workdir": ""
+    }
+}
+```
+
+[[reasoning]]
+== Reasoning
+
+This ITE proposes a change in the URI scheme of artifacts in in-toto metadata.
+The proposed change closely matches the URI structure laid out in RFC 3986. The
+URI structure propsed in that document is widely accepted and is versatile
+enough to allow for a wide variety of references. It is also easy to implement
+and extend support for due to the presence of a large number of standard
+libraries.
+
+[[backwards-compatibility]]
+== Backwards Compatibility
+
+If in-toto metadata is generated using an implementation of in-toto conforming
+to this ITE, verification using a non-conforming implementation will likely
+fail. It is still possible for verification to continue if the inspections don't
+use generic URIs. This is because the layout may contain inspections that use
+generic URIs.
+
+However, a conforming implementation should be capable of verifying in-toto
+metadata generated using a non-conforming implementation, as an ITE-4 conforming
+system must also conform to the actual in-toto specification.
+
+It's also possible two conforming systems are unable to verify the other's
+in-toto metadata as they're unaware of how to resolve certain URI tokens used by
+the other. This is again because of the possibility of inspections containing
+URIs that the other system is unable to resolve and calculate cryptographic
+information for.
+
+[[security]]
+== Security
+
+It's important to note that filepaths are a form of URIs too, and can be
+represented with the `scheme` `file://`. We have also discovered that `MATCH`
+and other artifact rules use only the URI strings themselves to find the
+appropriate entries in the link metadata. The changes proposed in this ITE,
+therefore, have no effect on the security guarantees made by the in-toto
+specification.
+
+Implementing systems that conform to this ITE must, however, be careful with how
+they compute cryptographic hashes of abstract entities.
```

Comment:
Hashing the contents of ITE-4-addressable artifacts is not in the scope of ITE-4?